internal/controller/storageclient_controller.go

```diff
@@ -447,12 +447,10 @@ func (r *storageClientReconcile) deletionPhase(externalClusterClient *providerCl
 		}
 	}
 
-	if res, err := r.offboardConsumer(externalClusterClient); err != nil {
-		r.log.Error(err, "Offboarding in progress.")
-	} else if !res.IsZero() {
-		// result is not empty
-		return res, nil
+	if _, err := r.offboardConsumer(externalClusterClient); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to offboard consumer for storageclient %v: %v", r.storageClient.Name, err)
```

Comment:
The function signature implies that a non-error requeue result might be returned from the `r.offboardConsumer` function. In that case, you cannot just ignore it and continue to remove the finalizer. 

There are 2 options here:
If the function can and might return a non-error requeue result then you need to hand it (similar to the original code). If it can't, the the Result object should be removed from the function signature.

Reply:
removed the result object from r.offboardConsumer()